code-challenges.test.js

```diff
@@ -10,45 +12,152 @@
 
-
+// 1a) Pseudo code:
+// Copy and paste test code template into assignment (make sure nothing is inside initial template)
+// Read problem statement and understand inputs/outputs (inputs need to be directly inserted into test code)
+// Create relevant info for describe, it, expect, and toEqual methods
+// Also, since the output will be random, need to include an expect.arrayContaining() statement inside .toEqual()
+// Test code output to ensure it is failing red (should get a reference error saying function is not defined)
+
+describe("mixItUp", () => {
+  it("A function which takes in an array, removes the first item from the array, and shuffles the remaining content", () => {
+    expect(mixItUp(["purple", "blue", "green", "yellow", "pink"])).toEqual(
+      expect.arrayContaining(["yellow", "blue", "pink", "green"])
+    );
+    expect(
+      mixItUp([
+        "chartreuse",
+        "indigo",
+        "periwinkle",
+        "ochre",
+        "aquamarine",
+        "saffron",
+      ])
+    ).toEqual(
+      expect.arrayContaining([
+        "saffron",
+        "aquamarine",
+        "periwinkle",
+        "indigo",
+        "ochre",
+      ])
+    );
+  });
+});
 
 // b) Create the function that makes the test pass.
 
-
+// 1b) Pseudo code:
+// Declare a function and name it mixItUp (const declaration)
+// Name the parameter as arr
+// Declare a variable named mixArray, and initalize it to the argument arr (let declaration)
+// Add .shift to mixArray
+// Create a for loop which iterates through all elements of mixArray
+// for loop logic:
+//      Create a variable to store a random value based on the mixArray length
+//      Create a while loop which checks to ensure the value stored at that random number's index is unique
+//      .push the mixArray[rando] into mixArrayNew
+// return mixArrayNew
+// Run the test to ensure it passes green
+
+const mixItUp = (arr) => {
+  let mixArray = arr;
+  mixArray.shift();
+  let mixArrayNew = [];
+  for (let i = 0; i < mixArray.length; i++) {
+    let rando = Math.floor(Math.random() * mixArray.length);
+    while (mixArrayNew.includes(mixArray[rando])) {
+      rando = Math.floor(Math.random() * mixArray.length);
+    }
+    mixArrayNew.push(mixArray[rando]);
+  }
+  return mixArrayNew;
+};
+
+// Note: I ran the intital test and got a red failure. Implementing the actual code took quite a bit of troubleshooting. I was initially using the length of mixArray - 1 for generating the value for rando, which when I ran the actual jest test, never actually finished. Taking that out fixed my issue. I know there is definitely potential for refactoring. I was trying to find a method which already shuffled the array, but I didn't find anything. If that was an option I would definitely use it. Also, I didn't realize you could change how jest tests code by adding methods inside of the expect values. I definitely need to look more into this.
```

Comment:
Nice work. This one is definitely tricky. Good job on troubleshooting. That is the key to the game.